tests/Mono.Android-Tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs

```diff
@@ -340,5 +340,83 @@ private static X509Certificate2 BuildClientCertificate ()
 
 			return clientCert;
 		}
+
+		[Test]
+		public async Task HttpContentStreamIsRewoundAfterCancellation ()
```

Comment:
This new test is passing:

<img width="1323" height="404" alt="Image" src="https://github.com/user-attachments/assets/6c3dca14-ec59-4221-bb41-cec579b9b43d" />